tasks/kernel_matrix_testing/infra.py

```diff
@@ -46,7 +48,7 @@ class RemoteCommandRunner:
     def run_cmd(ctx: Context, instance: 'HostInstance', cmd: str, allow_fail: bool, verbose: bool):
         res = ctx.run(
             cmd.format(
-                proxy_cmd=f"-o ProxyCommand='ssh -o StrictHostKeyChecking=no -i {instance.ssh_key} -W %h:%p ubuntu@{instance.ip}'"
+                proxy_cmd=f"-o ProxyCommand='ssh -o StrictHostKeyChecking=no -i {instance.ssh_key_path} -W %h:%p ubuntu@{instance.ip}'"
```

Comment:
suggestion: only include `-i` if `instance.ssh_key_path` is valid. Example: 
```
ssh_key = f"-i {instance.ssh_key_path}" if instance.ssh_key_path else ""
cmd.format(
    proxy_cmd=f"-o ProxyCommand='ssh -o StrictHostKeyChecking=no {ssh_key} -W %h:%p ubuntu@{instance.ip}'"
),
```

Reply:
Also, it might be helpful to print the entire command with the proxy information in the error message, if it fails.